lib/devices/MiioVacuum.js

```diff
@@ -70,49 +69,84 @@ class MiioVacuum {
 
     /**
      * Called when a message is received, either from cloud or local interface.
+     *
+     * @protected
      * @param {any} msg the json object sent by the remote device
-     * @returns boolean True if the message was handled.
+     * @returns {boolean} True if the message was handled.
      */
     onMessage(msg) {
         return false;
     }
 
-    /** Called once the dummycloud connection was established. */
+    /**
+     * Called once the dummycloud connection was established.
+     *
+     * @protected
+     */
     onCloudConnected() {
         Logger.info("Cloud connected");
         // start polling the map after a brief delay of 3.5s
         setTimeout(() => this.pollMap(), 3500);
     }
 
-    async pollStatus() {}
+    /**
+     * @protected
+     * @abstract
+     * @returns {Promise<void>}
+     */
+    async pollStatus() {
+        throw new NotImplementedError();
+    }
 
     /**
      * Returns the current status of the robot
+     *
+     * @public
```

Comment:
isn’t public the default already?

Reply:
Yes, public is the default. I simply added the access modifiers everywhere to make it clearer.